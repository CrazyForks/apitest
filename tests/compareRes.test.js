const { spwanTest } = require("./utils");

describe("compareRes", () => {
  test("general", async () => {
    const { stdout, code } = await spwanTest("compareRes/test.jsona", ["--ci"]);
    expect(code).toEqual(1);
    expect(stdout).toMatchSnapshot();
  });
  test("eval", async () => {
    const { stdout, code } = await spwanTest("compareRes/eval.jsona", ["--ci"]);
    expect(code).toEqual(1);
    expect(stdout).toMatchSnapshot();
  });
  test("type", async () => {
    const { stdout, code } = await spwanTest("compareRes/type.jsona", ["--ci"]);
    expect(code).toEqual(1);
    expect(stdout).toMatchSnapshot();
  });
  test("every", async () => {
    const { stdout, code } = await spwanTest("compareRes/every.jsona", ["--ci"]);
    expect(code).toEqual(1);
    expect(stdout).toMatchSnapshot();
  });
  test("some", async () => {
    const { stdout, code } = await spwanTest("compareRes/some.jsona", ["--ci"]);
    expect(code).toEqual(1);
    expect(stdout).toMatchSnapshot();
  });
  test("partial", async () => {
    const { stdout, code } = await spwanTest("compareRes/partial.jsona", ["--ci"]);
    expect(code).toEqual(0);
    expect(stdout).toMatchSnapshot();
  });
});
