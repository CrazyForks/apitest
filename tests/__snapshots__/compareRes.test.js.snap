// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`compareRes eval 1`] = `
"main.test1.req: is required at line 2 col 10
"
`;

exports[`compareRes every 1`] = `
"[1mmodule main[22m
[1m[22m[1m  @every[22m[32m âœ”[39m
[32m[39m[1m  @every throw[22m[31m âœ˜[39m
[31m[39m[1m  @every must be array value[22m[31m âœ˜[39m
[31m[39m
1. @every throw(main.test2)
[31m   main.test2.res.v1@every: some test fail[39m
[31m     main.test2.res.v1.0@eval: eval expr fail[39m
[31m[39m
[31m[39m2. @every must be array value(main.test3)
[31m   main.test3.res.v1@every: should have array value[39m
[31m[39m
[31m[39m"
`;

exports[`compareRes general 1`] = `
"[1mmodule main[22m
[1m[22m[1m  value not equal[22m[31m âœ˜[39m
[31m[39m[1m  nest value not equal[22m[31m âœ˜[39m
[31m[39m[1m  value type not equal[22m[31m âœ˜[39m
[31m[39m[1m  object prop length not equal[22m[31m âœ˜[39m
[31m[39m[1m  array elem length not equal[22m[31m âœ˜[39m
[31m[39m
1. value not equal(main.test1)
[31m   main.test1.res.v1: 2 â‰  3[39m
[31m[39m
[31m[39m2. nest value not equal(main.test2)
[31m   main.test2.res.v1.v2.v3: b â‰  a[39m
[31m[39m
[31m[39m3. value type not equal(main.test3)
[31m   main.test3.res.v1: type integer â‰  type string[39m
[31m[39m
[31m[39m4. object prop length not equal(main.test4)
[31m   main.test4.res.v1: keys length 1 â‰  2[39m
[31m[39m
[31m[39m5. array elem length not equal(main.test5)
[31m   main.test5.res.v1: elements length 1 â‰  2[39m
[31m[39m
[31m[39m"
`;

exports[`compareRes partial 1`] = `
"[1mmodule main[22m
[1m[22m[1m  @partial for object[22m[32m âœ”[39m
[32m[39m[1m  @partial for array[22m[32m âœ”[39m
[32m[39m
"
`;

exports[`compareRes some 1`] = `
"[1mmodule main[22m
[1m[22m[1m  @some[22m[32m âœ”[39m
[32m[39m[1m  @some throw[22m[31m âœ˜[39m
[31m[39m[1m  @some must be array value[22m[31m âœ˜[39m
[31m[39m
1. @some throw(main.test2)
[31m   main.test2.res.v1@some: no test pass[39m
[31m     main.test2.res.v1.0@eval: eval expr fail[39m
[31m     main.test2.res.v1.1@eval: eval expr fail[39m
[31m[39m
[31m[39m2. @some must be array value(main.test3)
[31m   main.test3.res.v1@some: should have array value[39m
[31m[39m
[31m[39m"
`;

exports[`compareRes type 1`] = `
"[1mmodule main[22m
[1m[22m[1m  @type[22m[32m âœ”[39m
[32m[39m[1m  @type null[22m[32m âœ”[39m
[32m[39m[1m  @type throw[22m[31m âœ˜[39m
[31m[39m
1. @type throw(main.test3)
[31m   main.test3.res.v1@type: type float â‰  type integer[39m
[31m[39m
[31m[39m"
`;
